### PR TITLE
Implementation of Themes

### LINKED ISSUE #9: [NEW FEAT.] Change theme to Dark/Light/System Default
The app doesn't have a dark theme or the toggle to switch between different modes.
Add a button and allow the user to switch themes.

### PR DESCRIPTION
**Pull request by Abh1noob**
# Closes #9 

# Changes Made:
- Created themeprovider.dart
  - Created Function for System, Dark and Light Modes.
- Created a dropdown box in about_view.dart and implemented themes through that box.

# Reference Images:
<table>
<tr>
<td>System Mode </td></tr>
<tr>
<td>
<table>
<tr><td>image</td>
<td>image</td>
</tr>
</table>
</td>
</tr>
<tr><td>Light Mode<td><tr>
<tr><td>image</td></tr>
</tr>
<tr><td>Dark Mode<td><tr>
<tr><td>image</td></tr>

### CODE REVIEW

File: lib/themeprovider.dart [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+  }
+  void setSystem() {
+    _value = ThemeMode.system;
+    notifyListeners();
+  }
+}
+
+
+
+// class ThemeSetter extends ChangeNotifier {
```
Comment (danger-ahead) [inline lib/themeprovider.dart:29]: Can you please remove the commented out lines if not required anymore?

If its required, please comment why its required in the code section itself.

**Review by danger-ahead — CHANGES_REQUESTED** [review]

**Comment by danger-ahead** [comment]
Also can you update the theme switch drop down button style? You could add some padding and maybe make it look more modern.

File: lib/themeprovider.dart (hunk 1 above)
Comment (Abh1noob) [reply lib/themeprovider.dart:29]: Ya sure, I forgot to remove it

**Comment by Abh1noob** [comment]
Where specifically you want me to place the button then? I'll make the button bit more modern thats fine

**Comment by Abh1noob** [comment]
> > Where specifically you want me to place the button then? I'll make the button bit more modern thats fine
> 
> I think placement is fine. Or at best if we could somehow place the a single icon button in the app bar, that would be the best.
> 
> The icon would show what theme the app is currently running in. For example a moon when it is in dark mode.
> 
> Once tapped, it can perhaps change to sun icon and the theme changes accordingly.
> 
> When tapped once more, the theme changes to system. And this cycle repeats.

Sure, I'll try. Also any specific icon you would like for 'System theme'?

**Comment by danger-ahead** [comment]
> > > Where specifically you want me to place the button then? I'll make the button bit more modern thats fine
> > 
> > I think placement is fine. Or at best if we could somehow place the a single icon button in the app bar, that would be the best.
> > 
> > The icon would show what theme the app is currently running in. For example a moon when it is in dark mode.
> > 
> > Once tapped, it can perhaps change to sun icon and the theme changes accordingly.
> > 
> > When tapped once more, the theme changes to system. And this cycle repeats.
> 
> Sure, I'll try. Also any specific icon you would like for 'System theme'?

Nothing as such. Choose what suits best.

**Comment by Abh1noob** [comment]
Check I have made changes

**Review by danger-ahead — APPROVED** [review]

**Comment by danger-ahead** [comment]
Nice work🙌

**Pull request merged**